WordPress/gutenberg
# [Full Site Editing]: Expand the templates that can be added - custom taxonomies, specific term, specific category and tag

## What?
Part of: https://github.com/WordPress/gutenberg/issues/37407

This PR expands the templates that can be added by handling custom taxonomies and specific terms of any public taxonomy, including `category` and `tag`.



## How?
It uses the same design for custom post types that is [already merged](https://github.com/WordPress/gutenberg/pull/41189). 

1. If we don't have `taxonomy-$taxonomy_slug` and we have available terms that don't have a specific template created, it will show a modal with two options to create either of them. You can search through the available terms to create a specific template for it.
2.  If we don't have `taxonomy-$taxonomy_slug` and we **don't** have available terms that don't have a specific template created, clicking the menu item will just create the `taxonomy-$taxonomy_slug`.
3. If we have `taxonomy-$taxonomy_slug` and we have available terms that don't have a specific template created, it will show the UI for  searching the available terms.
4. If we have `taxonomy-$taxonomy_slug` and we **don't** have available terms that don't have a specific template created, the menu item will not be added.

This also applies for `category` and `tag` with different slugs though(technical detail).



## Testing Instructions
1. In Site Editor's templates list create templates for specific categories/tags, etc.. or the general template and observe how it behaves
2. You can also register some custom taxonomies to test with



## Screenshots or screencast 

https://user-images.githubusercontent.com/16275880/175038882-31014ac2-0063-462b-bedf-d202fccf8833.mov
